File: ClientApp/src/api/authController.ts
import axios from "axios";
import { AuthModel } from "../models/AuthModel";
import { LoginModel } from "../models/LoginModel";
import { RegistrationModel } from "../models/RegistrationModel";
import authStore from "../stores/AuthStore";
import {EmployeeModel} from "../models/EmployeeModel";
import {EditEmployeeModel} from "../models/EditEmployeeModel";

const login = (model: LoginModel): Promise<AuthModel> =>
  axios.post("/api/auth/login", model)
    .then(response => response.data);

const register = (model: RegistrationModel): Promise<AuthModel> =>
  axios.post("/api/auth/register", model)
    .then(response => response.data)

const createEmployee = (model: RegistrationModel) =>
  axios.post("/api/Auth/CreateEmployee", model)

const editEmployee = (model: EditEmployeeModel) =>
  axios.put("/api/Auth/EditEmployee", model)

const getEmployees = (): Promise<EmployeeModel[]> =>
  axios.get("/api/Auth/Employees")
    .then(response => response.data)

const getEmployee = (id: number): Promise<EmployeeModel> =>
  axios.get(`/api/Auth/Employees/${id}`)
    .then(response => response.data)

const deleteEmployee = (id: number) =>
  axios.delete(`/api/Auth/DeleteEmployee/${id}`)

const logout = () =>
  axios.post("/api/auth/logout");

export default {
    login,
    register,
    getEmployees,
    getEmployee,
    createEmployee,
    editEmployee,
    deleteEmployee,
    logout
};

File: Controllers/AuthController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using KnowledgeControl.Models;
using KnowledgeControl.Services.Interfaces;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;

namespace KnowledgeControl.Controllers
{
  [ApiController]
  [Route("api/[controller]/[action]")]
  public class AuthController : ControllerBase
  {
    private readonly IAuthService _auth;

    public AuthController(IAuthService auth)
    {
      _auth = auth;
    }

    [HttpPost]
    public async Task Register(RegistrationModel model)
    {
      await _auth.Register(model);
    }

    [HttpPost]
    public async Task CreateEmployee(RegistrationModel model)
    {
      await _auth.RegisterEmployee(model);
    }

    [HttpPost]
    public async Task EditEmployee(EditEmployeeModel model)
    {
      await _auth.EditUser(model);
    }

    [HttpGet]
    public IEnumerable<EmployeeModel> GetEmployees() {
        return _auth.GetEmployees();
    }

    [HttpPost]
    public async Task<AuthModel> Login(LoginModel model) {
      var auth = await _auth.Authenticate(model.UserName, model.Password);
      
      Response.Cookies.Append(
        Options.AuthCookie, auth.Token, new CookieOptions { MaxAge = TimeSpan.FromDays(7) });
      Response.Cookies.Append(
        Options.IsEmployerCookie, 
        auth.IsEmployer ? "true" : "false", 
        new CookieOptions { MaxAge = TimeSpan.FromDays(7) });

      return auth;
    }
    
    [HttpPost]
    public StatusCodeResult LogOut()
    {
      Response.Cookies.Delete(Options.AuthCookie);
      Response.Cookies.Delete(Options.IsEmployerCookie);
      return Ok();
    }

    [HttpPost]
    public Task ChangePassword(EditPasswordModel passwords) =>
      _auth.ChangePassword(passwords.CurrentPassword, passwords.Password);
  }
}
